### PR TITLE
Fix the misleading diagnostic for `let_underscore_drop` on type without `Drop` implementation

### LINKED ISSUE #130430: Incorrect let_underscore_drop warning on type without Drop implementation
I tried this code:

```rust
#![deny(let_underscore_drop)]

fn main() {
    let _ = MyStruct {
        field: vec![1, 2, 3],
    };
}

struct MyStruct {
    field: Vec<usize>,
}
```

I got this warning:

```
error: non-binding let on a type that implements `Drop`
 --> src/main.rs:4:5
  |
4 | /     let _ = MyStruct {
5 | |         field: vec![1, 2, 3],
6 | |     };
  | |______^
  |
note: the lint level is defined here
 --> src/main.rs:1:9
  |
1 | #![deny(let_underscore_drop)]
  |         ^^^^^^^^^^^^^^^^^^^
help: consider binding to an unused variable to avoid immediately dropping the value
  |
4 |     let _unused = MyStruct {
  |         ~~~~~~~
help: consider immediately dropping the value
  |
4 ~     drop(MyStruct {
5 |         field: vec![1, 2, 3],
6 ~     });
  |
```

However, there is no `Drop` impl on the type, so the error message is misleading.
It is true however that there is a destructor, as defined here: https://doc.rust-lang.org/stable/std/ops/trait.Drop.html

That being said, it's insiginificant for all underlying types (only `#[may_dangle]` destructor of `Vec` and `Copy` type in the `Vec`), so ideally the warning still wouldn't even show.

### Meta

`rustc --version --verbose`:

```
rustc 1.80.1 (3f5fd8dd4 2024-08-06)
binary: rustc
commit-hash: 3f5fd8dd41153bc5fdca9427e9e05be2c767ba23
commit-date: 2024-08-06
host: x86_64-unknown-linux-gnu
release: 1.80.1
LLVM version: 18.1.7
```

<details><summary>Backtrace</summary>
<p>

```
   Compiling MinimalReproNonBindingLetDrop v0.1.0 (/home/elrendio/Stockly/MinimalReproNonBindingLetDrop)
warning: field `field` is never read
  --> src/main.rs:10:5
   |
9  | struct MyStruct {
   |        -------- field in this struct
10 |     field: Vec<usize>,
   |     ^^^^^
   |
   = note: `#[warn(dead_code)]` on by default

error: non-binding let on a type that implements `Drop`
 --> src/main.rs:4:5
  |
4 | /     let _ = MyStruct {
5 | |         field: vec![1, 2, 3],
6 | |     };
  | |______^
  |
note: the lint level is defined here
 --> src/main.rs:1:9
  |
1 | #![deny(let_underscore_drop)]
  |         ^^^^^^^^^^^^^^^^^^^
help: consider binding to an unused variable to avoid immediately dropping the value
  |
4 |     let _unused = MyStruct {
  |         ~~~~~~~
help: consider immediately dropping the value
  |
4 ~     drop(MyStruct {
5 |         field: vec![1, 2, 3],
6 ~     });
  |

warning: `MinimalReproNonBindingLetDrop` (bin "MinimalReproNonBindingLetDrop") generated 1 warning
error: could not compile `MinimalReproNonBindingLetDrop` (bin "MinimalReproNonBindingLetDrop") due to 1 previous error; 1 warning emitted
```

</p>
</details>

### PR DESCRIPTION
**Pull request by makai410**
Closes: #130430 
r? rust-lang/diagnostics

### CODE REVIEW

**Review by compiler-errors — APPROVED** [review]

File: src/tools/rust-analyzer/crates/ide-db/src/generated/lints.rs [hunk 1]
```diff
@@ -301,7 +301,7 @@ pub const DEFAULT_LINTS: &[Lint] = &[
     },
     Lint {
         label: "let_underscore_drop",
-        description: r##"non-binding let on a type that implements `Drop`"##,
+        description: r##"non-binding let on a type that has a destructor"##,
```
Comment (compiler-errors) [inline src/tools/rust-analyzer/crates/ide-db/src/generated/lints.rs:304]: Actually: please revert this change; afaict it's auto-generated, and rust-analyzer will deal with it on their own.

**Review by compiler-errors — CHANGES_REQUESTED** [review]

**Comment by compiler-errors** [comment]
@bors r-

Please rebase this into one commit; none of this PR's history should need to be synced next time that rust-analyzer does a subtree sync.

**Comment by compiler-errors** [comment]
Or you can do (making sure you're on this branch):

```
git reset HEAD~2
git commit -am "Fix the misleading diagnostic for let_underscore_drop on type without Drop implementation"
# then push with `--force-with-lease`
```

**Comment by fee1-dead** [comment]
> Please rebase this into one commit;

Oops, thanks for catching this! Wasn't looking at the commits log when I reviewed..

**Pull request merged**